src/ports/linux/pnal_filetools.c

```diff
@@ -0,0 +1,215 @@
+ *                                Terminated string.
+ *                                Might be modified also on failure.
+ * @param size             In :   Size of the outputbuffer.
+ *
+ * @return 0 on success.
+ *         -1 if an error occured.
+ */
+static int pnal_locate_script (const char * scriptname, char * path, size_t size)
+{
+   uint16_t ix = 0;
+   char directory_for_binary
+      [PNET_MAX_DIRECTORYPATH_LENGTH + PNET_MAX_FILENAME_LENGTH] = {0};
+
+   if (strlen (scriptname) == 0)
+   {
+      LOG_ERROR (PF_PNAL_LOG, "PNAL(%d): No script name given.\n", __LINE__);
+      return -1;
+   }
+
+   /* Look in the same directory as the main binary */
+   if (
+      pnal_get_directory_of_binary (
+         directory_for_binary,
+         sizeof (directory_for_binary)) != 0)
+   {
+      LOG_ERROR (
+         PF_PNAL_LOG,
+         "PNAL(%d): Failed to locate the directory of main binary.\n",
+         __LINE__);
+      return -1;
+   }
+   pf_file_join_directory_filename (
+      directory_for_binary,
+      scriptname,
+      path,
+      size);
+   if (pnal_does_file_exist (path))
+   {
+      return 0;
+   }
+
+   /* Look in list of typical PATH directories */
+   for (ix = 0; ix < NELEMENTS (script_search_directories); ix++)
+   {
+      pf_file_join_directory_filename (
+         script_search_directories[ix],
+         scriptname,
+         path,
+         size);
+      if (pnal_does_file_exist (path))
+      {
+         return 0;
+      }
+   }
+
+   /* Look in current directory */
+   pf_file_join_directory_filename (NULL, scriptname, path, size);
+   if (pnal_does_file_exist (path))
+   {
+      return 0;
+   }
+
+   return -1;
+}
+
+int pnal_execute_script (int argc, const char * argv[])
+{
+   uint16_t ix;
+   char path[PNET_MAX_DIRECTORYPATH_LENGTH + PNET_MAX_FILENAME_LENGTH] = {0};
+   char * scriptarg[argc + 1];
+   char * scriptenviron[] = {NULL};
+   int script_returnvalue = 0;
+   int childstatus = 0;
+   pid_t pid;
+
+   if (argc < 1)
+   {
+      LOG_ERROR (PF_PNAL_LOG, "PNAL(%d): No script name given.\n", __LINE__);
+      return -1;
+   }
+
+   if (pnal_locate_script (argv[0], path, sizeof (path)) != 0)
+   {
+      LOG_ERROR (
+         PF_PNAL_LOG,
+         "PNAL(%d): Failed to locate the script %s.\n",
+         __LINE__,
+         argv[0]);
+      return -1;
+   }
+
+#if LOG_DEBUG_ENABLED(PNET_LOG)
+   printf ("PNAL(%d): Command for script: %s", __LINE__, path);
+   for (ix = 1; ix < argc; ix++)
+   {
+      printf (" '%s'", argv[ix]);
+   }
+   printf ("\n");
+#endif
+
+   /* Prepare null terminated argument vector */
+   for (ix = 0; ix < NELEMENTS (scriptarg); ix++)
+   {
+      scriptarg[ix] = (char *)argv[ix];
+   }
+   scriptarg[argc] = NULL;
+
+   /* Fork and exec */
+   pid = fork();
+   if (pid < 0)
+   {
+      LOG_ERROR (
+         PF_PNAL_LOG,
+         "PNAL(%d): Failed to fork the process.\n",
+         __LINE__);
+      return -1;
+   }
+   if (pid == 0)
```

Comment:
Maybe a comment or two on that the pid return value defines if execution is in child or parent is more useful than the comment on line 194.

Reply:
Done.